### PR TITLE
Skinny Dipper Script Changes: modify mint_multiple_tokens, sign_all to be faster

### PR DESCRIPTION
**Pull request by swimalick**
These are a bunch of hacks I made to make minting and signing faster. I basically made everything async. 

Not really recommended to use these because they can really drain your wallet. But in case we ever have more collectibles in the future, we can do this again

### CODE REVIEW

File: js/packages/cli/src/commands/mint.ts [hunk 1]
```diff
@@ -399,7 +399,7 @@ export async function mintV2(
     (data.gatekeeper ? 33 : 0) +
     (data.gatekeeper?.expireOnUse ? 66 : 0);
 
-  log.info('Transaction size estimate: ', txnEstimate);
+  //log.info('Transaction size estimate: ', txnEstimate);
```
Comment (goriunov) [inline js/packages/cli/src/commands/mint.ts:402]: Same here ! delete the code, but I think this log is quite important or debugging probably better put it as debug

File: js/packages/cli/src/commands/signAll.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ import { AccountAndPubkey } from '../helpers/accounts';
 import { signMetadataInstruction } from './sign';
 import log from 'loglevel';
 import { sleep } from '../helpers/various';
+import * as fs from 'fs';
```
Comment (goriunov) [inline js/packages/cli/src/commands/signAll.ts:22]: Why do you need this import !

File: js/packages/cli/src/helpers/constants.ts [hunk 3]
```diff
@@ -100,7 +100,7 @@ export const CONFIG_LINE_SIZE = 4 + 32 + 4 + 200;
 
 export const CACHE_PATH = './.cache';
 
-export const DEFAULT_TIMEOUT = 15000;
+export const DEFAULT_TIMEOUT = 150000;
```
Comment (goriunov) [inline js/packages/cli/src/helpers/constants.ts:103]: That is not good idea... just increase it to 60s or something I think that should be enough.

File: js/packages/cli/src/helpers/transactions.ts [hunk 4]
```diff
@@ -118,7 +118,8 @@ export async function sendSignedTransaction({
   } catch (err) {
     log.error('Timeout Error caught', err);
     if (err.timeout) {
-      throw new Error('Timed out awaiting confirmation on transaction');
+      log.error('timed out but continuing');
+      //throw new Error('Timed out awaiting confirmation on transaction, timeout is ' + DEFAULT_TIMEOUT);
```
Comment (goriunov) [inline js/packages/cli/src/helpers/transactions.ts:122]: Not sure on what i think about this one..  but note that this file is used across other things where if a transaction failed it must exist and this change is braking behaviour

File: js/packages/cli/src/candy-machine-v2-cli.ts [hunk 5]
```diff
@@ -1002,16 +1002,27 @@ programCommand('mint_multiple_tokens')
     log.info(`Minting ${NUMBER_OF_NFTS_TO_MINT} tokens...`);
 
     const mintToken = async index => {
+		  var max = 30; //seconds
+		  var min = 1; //seconds
+		  var rand = Math.floor(Math.random() * (max - min + 1) + min); //Generate Random number between 5 - 10
+		  log.info(`waiting ${rand} seconds`)
+		  await new Promise(f => setTimeout(f, rand * 1000)); // Wait random seconds to not get 429 too many requests
```
Comment (goriunov) [inline js/packages/cli/src/candy-machine-v2-cli.ts:1009]: 429 should not be the case for private RPC, I don't think it makes sense to handle this, you also want to make the batch size configurable

File: js/packages/cli/src/commands/signAll.ts [hunk 6]
```diff
@@ -248,18 +257,55 @@ async function getCandyMachineVerifiedMetadata(
...
   let total = 0;
+  let counter = 0;
+
+  var size = batchsize;
+  var arrayOfArrays = [];
+  for (var i=0; i<metadataList.length; i+=size) {
+       arrayOfArrays.push(metadataList.slice(i,i+size));
+  }
+  //console.log(arrayOfArrays);
+  for (var i = 0; i < arrayOfArrays.length; i += 100) {
```
Comment (goriunov) [inline js/packages/cli/src/commands/signAll.ts:268]: There are cleaner ways to do that...

**Comment by swimalick** [comment]
Whoops I meant to only make this PR in our forked repo 😓 thanks for the review

**Review by Pearlminima — APPROVED** [review]

**Pull request closed without merging**